musicbot/playlist.py

```diff
@@ -344,11 +344,17 @@ def peek(self):
         """
             (very) Roughly estimates the time till the queue will 'position'
         """
-        estimated_time = sum(e.duration for e in islice(self.entries, position - 1))
+        if any(e.duration for e in islice(self.entries, position - 1) == None):
+            raise InvalidDataError('no duration data')
+        else:
+            estimated_time = sum(e.duration for e in islice(self.entries, position - 1))
 
         # When the player plays a song, it eats the first playlist item, so we just have to add the time back
         if not player.is_stopped and player.current_entry:
-            estimated_time += player.current_entry.duration - player.progress
+            if player.current_entry.duration == None: # duration can be 0
+                raise InvalidDataError('no duration data in current entry')
```

Comment:
We shouldn't return an error if no duration is found. Would be much more user friendly to return a string stating that no duration is found because its not an error, its an expected condition.